src/coreclr/vm/profilinghelper.cpp

```diff
@@ -806,23 +806,28 @@ HRESULT ProfilingAPIUtility::AttemptLoadProfilerList()
         return S_OK;
     }
 
-    WCHAR *pOuter = NULL;
-    WCHAR *pInner = NULL;
-    WCHAR *currentSection = NULL;
-    WCHAR *currentPath = NULL;
-    WCHAR *currentGuid = NULL;
+    SString profilerList{wszProfilerList};
 
     HRESULT storedHr = S_OK;
-    // Get each semicolon delimited config
-    currentSection = wcstok_s(wszProfilerList, W(";"), &pOuter);
-    for (;currentSection != NULL; currentSection = wcstok_s(NULL, W(";"), &pOuter))
+    for (SString::Iterator sectionStart = profilerList.Begin(), sectionEnd = profilerList.Begin();
+        profilerList.Find(sectionEnd, W(';'));
+        sectionStart = ++sectionEnd)
     {
-        // Parse this config "path={guid}"
-        currentPath = wcstok_s(currentSection, W("="), &pInner);
-        currentGuid = wcstok_s(NULL, W("="), &pInner);
-
+        SString::Iterator pathEnd = sectionStart;
+        if (!profilerList.Find(pathEnd, W('=')) || pathEnd > sectionEnd)
+        {
+            ProfilingAPIUtility::LogProfError(IDS_E_PROF_BAD_PATH);
+            storedHr = E_FAIL;
+            continue;
+        }
+        SString::Iterator clsidStart = pathEnd + 1;
+        profilerList.Replace(pathEnd, W('\0'));
+        profilerList.Replace(sectionEnd, W('\0'));
```

Comment:
This hack with inserting null terminators does not look great.

If you want to use SString, can you create the proper substring using the SString constructor that takes start & end?